proposed/2022/DotnetListPackageMachineReadableJsonOutput.md

```diff
@@ -772,6 +772,139 @@ Outputs json for format version 1, if it's not specified then latest version'll
 }
 ```
 
+### Error handling
+
+In case of an error it would be written into `problems` section. But in case there is parameter, options error then it may defaults back to console output instead of json output.
+Also in case of runtime there's error then it may not show json output at all because it needs all the data for json output before display.
+
+#### `> dotnet list package`
+
+`MyProjectB.csproj` was not restored.
+
+```dotnetcli
+Project 'MyProjectA' has the following package references
+   [netcoreapp3.1]:
+   Top-level Package                      Requested             Resolved
+   > Microsoft.Extensions.Primitives      [1.0.0, 5.0.0]        1.0.0
+   > NuGet.Commands                       4.8.0-preview3.5278   4.8.0-preview3.5278
+   > Text2Xml.Lib                         [1.1.2, 2.0.0)        1.1.2
+
+No assets file was found for `C:\Users\userA\repos\MainApp\src\lib\MyProjectB.csproj`. Please run restore before running this command.
+```
+
+#### `> dotnet list package --format json`
+
+```json
+{
+  "version": 1,
+  "parameters": "",
+  "problems": [
+    "No assets file was found for `C:/Users/userA/repos/MainApp/src/lib/MyProjectB.csproj`. Please run restore before running this command."
+  ],
```

Comment:
should it be an array of objects, rather than array of strings?

```json
[
  {
    "project": "C:/Users/userA/repos/MainApp/src/lib/MyProjectB.csproj",
    "message": "No assets file was found for `C:/Users/userA/repos/MainApp/src/lib/MyProjectB.csproj`. Please run restore before running this command."
  },
  {
    "project": "C:/whatever/my.csproj",
    "source":  "https://private/index.json",
    "message": "HTTP timeout after 100 seconds"
 }
]
```

Reply:
Agree strongly with this - if the different _kinds_ of errors are known or can be anticipated, consider also having a `code` field for these problem structures - a code is often something that can be searched on, and often can be put into urls for easy user lookup: `aka.ms/nuget/listpackage#errors_<code>` for example.  This isn't a 100% match, but the HTTP Problem Details RFC ([RFC 7807](https://www.rfc-editor.org/rfc/rfc7807#section-3.1)) is a good example of this (where the `type` field maps to the `code` field I describe).

Reply:
@zivkan @baronfel 
I added array of objects instead of array of strings into spec. Please review latest spec.